src/test/unit/PaymentCoordinatorUnit.t.sol

```diff
@@ -0,0 +1,1613 @@
+            uint256[] memory earnings = _getCumulativeEarnings(claim);
+            uint256[] memory tokenBalancesBefore = _getClaimTokenBalances(claimer, claim);
+
+            _assertPaymentClaimedEvents(root, claim);
+            paymentCoordinator.processClaim(claim);
+
+            uint256[] memory tokenBalancesAfter = _getClaimTokenBalances(claimer, claim);
+
+            for (uint256 i = 0; i < totalClaimedBefore.length; ++i) {
+                assertEq(earnings[i], totalClaimedBefore[i], "totalClaimed should not be incremented");
+
+                assertEq(tokenBalancesBefore[i], tokenBalancesAfter[i], "token balance should not be incremented");
+            }
+
+            cheats.stopPrank();
+        }
+    }
+
+    /// @notice Claim against latest submitted root, rootIndex 3 but modify some of the leaf values
+    /// forge-config: default.fuzz.runs = 10
+    function testFuzz_processClaim_Revert_WhenInvalidTokenClaim(
+        bool setClaimerFor,
+        address claimerFor
+    ) public filterFuzzedAddressInputs(claimerFor) {
+        // if setClaimerFor is true, set the earners claimer to the fuzzed address
+        address claimer;
+        if (setClaimerFor) {
+            cheats.prank(earner);
+            paymentCoordinator.setClaimerFor(earner, claimerFor);
+            claimer = claimerFor;
+        } else {
+            claimer = earner;
+        }
+
+        // Parse all 3 claim proofs for distributionRoots 0,1,2 respectively
+        IPaymentCoordinator.PaymentMerkleClaim[] memory claims = _parseAllProofs();
+        IPaymentCoordinator.PaymentMerkleClaim memory claim = claims[2];
+
+        uint32 rootIndex = claim.rootIndex;
+        (, , uint64 activatedAt) = paymentCoordinator.distributionRoots(rootIndex);
+        cheats.warp(activatedAt);
+
+        // Modify Earnings
+        claim.tokenLeaves[0].cumulativeEarnings = 1e20;
+        claim.tokenLeaves[1].cumulativeEarnings = 1e20;
+
+        // Check claim is not valid from both checkClaim() and processClaim() throwing a revert
+        cheats.startPrank(claimer);
+
+        cheats.expectRevert("PaymentCoordinator._verifyTokenClaim: invalid token claim proof");
+        assertFalse(paymentCoordinator.checkClaim(claim), "PaymentCoordinator.checkClaim: claim not valid");
+
+        cheats.expectRevert("PaymentCoordinator._verifyTokenClaim: invalid token claim proof");
+        paymentCoordinator.processClaim(claim);
+
+        cheats.stopPrank();
+    }
+
+    /// @notice Claim against latest submitted root, rootIndex 3 but modify some of the leaf values
+    /// forge-config: default.fuzz.runs = 10
+    function testFuzz_processClaim_Revert_WhenInvalidEarnerClaim(
+        bool setClaimerFor,
+        address claimerFor,
+        address invalidEarner
+    ) public filterFuzzedAddressInputs(claimerFor) {
+        // if setClaimerFor is true, set the earners claimer to the fuzzed address
+        address claimer;
```

Comment:
putting the starting block (up to 1513) into an internal, reused function would be good